XLS-0056d-batch/README.md

```diff
@@ -0,0 +1,626 @@
+
+#### 2.2.1. `ALLORNOTHING`
+All or nothing. All transactions must succeed for any of them to succeed.
+
+#### 2.2.2. `ONLYONE`
+The first transaction to succeed will be the only one to succeed; all other transactions either failed or were never tried.
+
+While this can sort of be done by submitting multiple transactions with the same sequence number, there is no guarantee that the transactions are processed in the same order they are sent.
+
+#### 2.2.3. `UNTILFAILURE`
+All transactions will be applied until the first failure, and all transactions after the first failure will not be applied.
+
+#### 2.2.4. `INDEPENDENT`
+All transactions will be applied, regardless of failure.
+
+### 2.3. `RawTransactions`
+
+`RawTransactions` contains the list of transactions that will be applied. There can be up to 8 transactions included. These transactions can come from one account or multiple accounts.
+
+Each inner transaction:
+* **Must** contain a `BatchTxn` field (see section 3 for details).
+* **Must not** have a sequence number. It must use a sequence number value of `0`.
+* **Must not** have a fee. It must use a fee value of `"0"`.
+* **Must not** be signed (the global transaction is already signed by all relevant parties). They must instead have an empty string (`""`) in the `SigningPubKey` and `TxnSignature` fields.
+
+A transaction will be considered a failure if it receives any result that is not `tesSUCCESS`.
+
+**This field is not included in the validated transaction, nor is it used to compute the outer transaction signature(s)**, since all transactions are included separately as a part of the ledger.
+
+### 2.4. `TxnIDs`
+
+`TxnIDs` contains a list of the transaction hashes/IDs for all the transactions contained in `RawTransactions`. This is the only part of the inner transactions that is saved as a part of the ledger within the `Batch` transaction, since the inner transactions themselves will be their own transactions on-ledger. The hashes in `TxnIDs` **must** be in the same order as the raw transactions in `RawTransactions`.
+
+While this field seems complicated/confusing to work with, it can easily be abstracted away (e.g. as a part of autofilling) in tooling, and it's easy for `rippled` to check a hash doesn't match its corresponding transaction in `RawTransaction`.
+
+### 2.5. `BatchSigners`
+
+This field operates similarly to [multisign](https://xrpl.org/docs/concepts/accounts/multi-signing/) on the XRPL. It is only needed if multiple accounts' transactions are included in the `Batch` transaction; otherwise, the normal transaction signature provides the same security guarantees.
+
+Every account that has at least one inner transaction, excluding the outer account (if applicable), must have a `BatchSigners` field.
+
+|FieldName | Required? | JSON Type | Internal Type |
+|:---------|:-----------|:---------------|:------------|
+|`Account`|✔️|`string`|`STAccount`|
+|`SigningPubKey`| |`string`|`STBlob`|
+|`Signature`| |`string`|`STBlob`|
```

Comment:
Wouldn't it be easier to understand if the field name is the same as `TxnSignature` used as CommonField?

Reply:
It would technically have to be `TxnsSignature`, so I'd prefer leaving it as `Signature`. But it's not a strong opinion.

Reply:
If we use `Signer` the the sign checks that currently exist would need to be updated/duplicated. So i'm using `TxnSignature`.

Reply:
@dangell7 just to confirm: I should update the field here to be `TxnSignature`?

Reply:
yes please

Reply:
We are using `BatchSigner`. Thats whats in the spec I think (JSON). I guess we could do BatchSigners/Signer and add Signers as an optional on the Signer object...  if thats what you mean.

Above I said `Signer` but meant `Signature` we are using `TxnSignature` so that the signing/verification was the same between the two (signle/multi) but I think I could use `Signature` for just the BatchSigner and leave the signers the same. 

Up to you.